lib/middleware/validateSession.js

```diff
@@ -0,0 +1,70 @@
+var getSessionVal = function(obj, key) {
+  key = key.split('.');
+
+  for (var i = 0; i < key.length; i++) {
+    var curKey = key[i];
+
+    if (curKey in obj) {
+      obj = obj[curKey];
+    } else {
+      return null;
+    }
+  }
+
+  return obj;
+};
+
+module.exports = function validateSession(mediator, mbaasApi, opts) {
+
+  return function validateSessionMiddleware(req, res, next) {
+    var session,
+      sessionToken;
+
+    opts = {
```

Comment:
I think you can make the assumption that the session ID will be stored in the feedhenry params or there isn't one.

This module is not going to be used anywhere other than a RHMAP cloud app.

I don't think you need the `getSessionVal` function.

```
var fhParams = req.fh_params || {__fh: {}};
var sessionToken = fhParams.__fh.sessionToken || fhParams.__fh.sessiontoken;
```